src/client.ts

```diff
@@ -203,27 +203,28 @@ export class MqttClient extends EventEmitter {
     }
   }
 
-  private waitForConnack(): Promise<IConnackPacket> {
-    return new Promise((res, rej) => {
-      const connectionTimeout = () => {
-        rej('CONNECTION TIMEOUT')
-      }
-      const connackTimer = setTimeout(connectionTimeout, this._options.connectTimeout)
-      this.on('connack', (connackPacket) => {
-        this.removeListener('connack', res)
-        clearTimeout(connackTimer)
-        res(connackPacket)
+  private _awaitConnack(): Promise<IConnackPacket> {
+    return new Promise((resolve, reject) => {
+      const connackTimeout = setTimeout(

```

Comment:
maybe we should create a `sleep` method that's a promisified `setTimeout` so that our code looks less callbacky here.

Reply:
Yeah, I think we're going to have timeouts in other places too so good call